spec/value_inspector_spec.rb

```diff
@@ -94,25 +165,31 @@
 
     it "detects all elements" do
       expect(schema).to eq(
+        class: "Hash",
         properties: [
           {
             :class => "Array",
             :name => "items",
             :properties => [
-              { :class => "Integer", :name => "id" },
-              { :class => "String", :name => "category" },
-              { :class => "String", :name => "chargebee_plan_id" },
-              { :class => "String", :name => "country" },
-              { :class => "String", :name => "created_at" },
-              { :class => "TrueClass", :name => "is_current" },
-              { :class => "FalseClass", :name => "is_insurance" },
-              { :class => "TrueClass", :name => "is_public" },
-              { :class => "String", :name => "offering_key" },
-              { :class => "NilClass", :name => "plan_type" },
-              { :class => "Integer", :name => "price" },
-              { :class => "NilClass", :name => "stripe_plan_id" },
-              { :class => "NilClass", :name => "stripe_price_id" },
-              { :class => "String", :name => "updated_at" },
+              {
+                :class => "Hash",
+                :properties => [
+                  { :class => "Integer", :name => "id" },
+                  { :class => "String", :name => "category" },
+                  { :class => "String", :name => "chargebee_plan_id" },
+                  { :class => "String", :name => "country" },
+                  { :class => "String", :name => "created_at" },
+                  { :class => "TrueClass", :name => "is_current" },
+                  { :class => "FalseClass", :name => "is_insurance" },
+                  { :class => "TrueClass", :name => "is_public" },
+                  { :class => "String", :name => "offering_key" },
+                  { :class => "NilClass", :name => "plan_type" },
+                  { :class => "Integer", :name => "price" },
+                  { :class => "NilClass", :name => "stripe_plan_id" },
+                  { :class => "NilClass", :name => "stripe_price_id" },
+                  { :class => "String", :name => "updated_at" }
+                ]
+              },
             ],
```

Comment:
As far as backwards compatibility goes, I believe this is the only concerning change. Objects within an array now describe their type and any properties within a new object.

Reply:
Tested this with Forem.  `openapi.yml` represented objects within an array with their type and any properties within a new object.

```
@@ -47632,8 +47905,18 @@ paths:
                       body_markdown:
                         type: array
+                        items:
+                          type: object
+                          properties:
+                            undefined:
+                              type: string
                       listing_category:
                         type: array
+                        items:
+                          type: object
+                          properties:
+                            undefined:
+                              type: string
```

```
@@ -47755,6 +48036,11 @@ paths:
                     properties:
                       listing_category:
                         type: array
+                        items:
+                          type: object
+                          properties:
+                            undefined:
+                              type: string
                   status:
```

Reply:
The `undefined` here is a problem.